test/system/src/main/java/io/pravega/test/system/framework/Utils.java

```diff
@@ -116,10 +119,36 @@ public static Service createPravegaSegmentStoreService(final URI zkUri, final UR
                 return  new PravegaSegmentStoreDockerService(serviceId, zkUri, hdfsUri, contUri);
             case KUBERNETES:
             default:
-                return new PravegaSegmentStoreK8sService(serviceId, zkUri);
+                return new PravegaSegmentStoreK8sService(serviceId, zkUri, getPravegaProperties());
+        }
+    }
+
+    public static PravegaProperties getPravegaProperties() {
+        if (!SECURITY_ENABLED) {
+            return PravegaProperties.builder().build();
+        } else {
+            return PravegaProperties.builder().buildWithSecurity();
         }
     }
 
+    public static ClientConfig getClientConfig(URI controllerUri) {
+        if (!SECURITY_ENABLED) {
+            log.info("Generating config with auth disabled.");
+            return ClientConfig.builder().controllerURI(controllerUri).build();
+        } else {
+            log.info("Generating config with auth enabled.");
+            return ClientConfig.builder()
+                               // auth
+                               .credentials(new DefaultCredentials("1111_aaaa", "admin"))
```

Comment:
Can the credential data be overridden through environment variables and/ JVM/system properties? There should be a way to do that.

Reply:
Taking credentials through environment variables is already supported by ClientConfigBuilder. Could use that builder when required.

Reply:
Sounds good.

Reply:
The credentials will be read from the env/ system properties only if the `credentials`  is not explicitly set.
ref: `io.pravega.client.ClientConfig.ClientConfigBuilder#extractCredentials` 
We could always read the credentials via the environment, right? This would avoid the hardcoding of credentials in the framework.

Reply:
Taking credentials through environment variables would require passing these variables ( both username and password) in clear text via gradle at run time. This is not very different from directly hardcoding it in the code.

Reply:
To make it truly configurable should be read from a properties file supplied at runtime. This can be done if needed at a later stage when we have the need to support mulitple security environments.

Reply:
ok.